### PR TITLE
revised upgrade documentation

### PR DESCRIPTION
**Pull request by dmitsh**
revised upgrade documentation

### CODE REVIEW

File: examples/k8s-upgrade/README.md [hunk 1]
```diff
@@ -2,18 +2,116 @@
...
+ - create new VM and install desired orchestrator version
+ - add the new VM to the cluster
 
-The cluster definition file examples demonstrate initial cluster configurations:
-- **kubernetes1.5.json** - Kubernetes cluster v1.5 with Linux agent pool
-- **kubernetes1.5-win.json** - Kubernetes cluster v1.5 with Windows agent pool
-- **kubernetes1.5-hybrid.json** - Kubernetes cluster v1.5 with Linux and Windows agent pools
-- **kubernetes1.6.json** - Kubernetes cluster v1.6 with Linux agent pool
-- **kubernetes1.7.json** - Kubernetes cluster v1.7 with Linux agent pool
+*acs-engine* allows a subsequent minor version upgrade at a time, for example, from 1.5.x to 1.6.x.
```
Comment (jackfrancis) [inline examples/k8s-upgrade/README.md]: substitute "a subsequent minor version" with "one subsequent minor version"

File: examples/k8s-upgrade/README.md [hunk 2]
```diff
@@ -2,18 +2,116 @@
...
 
-The ***.env** files are used to set desired kubernetes version and instruct test framework to invoke post-deploy instructions implemented in **k8s-upgrade.sh** script.
+For upgrade that spans over more than a single minor version, this operation should be called several times, each time advancing the minor version by one. For example, to upgrade from 1.6.* to 1.8.* one should first upgrade the cluster to 1.7.*, following by upgrading it to 1.8.*
```
Comment (weinong) [inline examples/k8s-upgrade/README.md]: grammar: followed by upgrading it to 1.8.*
minor improvement: can you put code block markdown around x.y.z for better readability?

File: examples/k8s-upgrade/README.md (hunk 1 above)
Comment (weinong) [reply examples/k8s-upgrade/README.md]: Do you think it's useful or not to explain how master and agent nodes got upgrade?

Comment (dmitsh) [reply examples/k8s-upgrade/README.md]: @weinong lines 8-12 don't do that?

**Pull request merged**